collated/main/raisa2010.md

```diff
@@ -0,0 +1,1015 @@
+# raisa2010
+###### /java/seedu/address/logic/commands/tasks/AddTaskCommand.java
+``` java
+/**
+ * Adds a task to the task manager
+ */
+public class AddTaskCommand extends UndoableCommand {
+
+    public static final String COMMAND_WORD = "add";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a task to the task manager. "
+            + "Parameters: "
+            + "DESCRIPTION "
+            + PREFIX_STARTDATE + " START DATE "
+            + PREFIX_DEADLINE_BY + "/" + PREFIX_DEADLINE_ON + " DEADLINE DATE "
+            + PREFIX_TAG + "TAG";
+
+    public static final String MESSAGE_SUCCESS = "Task has been added: %1$s";
+    public static final String MESSAGE_DUPLICATE_TASK = "This task already exists in the task manager";
+
+    private final Task toAdd;
+
+    /**
+     * Creates an AddTaskCommand to add the specified {@code ReadOnlyTask}
+     */
+    public AddTaskCommand(ReadOnlyTask task) {
+        toAdd = new Task(task);
+    }
+
+    @Override
+    public CommandResult executeUndoableCommand() throws CommandException {
+        requireNonNull(model);
+        try {
+            model.addTask(toAdd);
+            return new CommandResult(String.format(MESSAGE_SUCCESS, toAdd));
+        } catch (DuplicateTaskException e) {
+            throw new CommandException(MESSAGE_DUPLICATE_TASK);
+        }
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+                || (other instanceof AddTaskCommand // instanceof handles nulls
+                && toAdd.equals(((AddTaskCommand) other).toAdd));
+    }
+}
+```
+###### /java/seedu/address/logic/Logic.java
+``` java
+    /** Returns an unmodifiable view of the filtered list of tasks */
+    ObservableList<ReadOnlyTask> getFilteredTaskList();
+
+```
+###### /java/seedu/address/logic/parser/AddTaskCommandParser.java
+``` java
+/**
+ * Parses input arguments and creates a new AddTaskCommand object
+ */
+public class AddTaskCommandParser implements Parser<AddTaskCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddTaskCommand
+     * and returns an AddTaskCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public AddTaskCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        ArgumentMultimap argMultimap =
```

Comment:
it is not really recommended to break after "=" because it makes the next line seems like a standalone statement.